packages/plugins/src/hardhat/tasks.ts

```diff
@@ -148,6 +158,177 @@ subtask(TASK_CHUGSPLASH_FETCH)
+        config.options.owner
+      )
+
+      spinner.succeed('Project successfully created')
+    }
+  )
+
+task(TASK_CHUGSPLASH_LIST_ALL_PROJECTS)
+  .setDescription('Lists all existing ChugSplash projects')
+  .setAction(async () => {
+    spinner.start('Getting list of all projects...')
+
+    const provider = new providers.JsonRpcProvider()
+    const ChugSplashRegistry = new Contract(
+      CHUGSPLASH_REGISTRY_ADDRESS,
+      ChugSplashRegistryABI,
+      provider
+    )
+
+    const events = await ChugSplashRegistry.queryFilter(
+      ChugSplashRegistry.filters.ChugSplashProjectRegistered()
+    )
+
+    spinner.stop()
+    events.forEach((event) =>
+      console.log(
+        `Project: ${event.args.nonIndexedProjectName}\t\tManager: ${event.args.manager}`
+      )
+    )
+  })
+
+task(TASK_CHUGSPLASH_PROPOSE)
+  .setDescription('Proposes a new ChugSplash bundle')
+  .addParam('bundleHash', 'hash of the bundle')
+  .addParam('deployConfig', 'path to chugsplash deploy config')
+  .setAction(
+    async (
+      args: {
+        bundleHash: string
+        deployConfig: string
+      },
+      hre
+    ) => {
+      spinner.start('Proposing the bundle...')
+
+      const config: ChugSplashConfig = await hre.run(TASK_CHUGSPLASH_LOAD, {
+        deployConfig: args.deployConfig,
+      })
+
+      const provider = new providers.JsonRpcProvider()
+      const ChugSplashRegistry = new Contract(
+        CHUGSPLASH_REGISTRY_ADDRESS,
+        ChugSplashRegistryABI,
+        await provider.getSigner()
+      )
+
+      const bundle: ChugSplashActionBundle = await hre.run(
+        TASK_CHUGSPLASH_BUNDLE_LOCAL,
+        {
+          deployConfig: args.deployConfig,
+        }
+      )
+
+      await ChugSplashRegistry.proposeChugSplashBundle(
+        config.options.name,
+        args.bundleHash,
+        bundle.actions.length,
+        args.deployConfig
+      )
+      spinner.succeed('Bundle successfully proposed')
+    }
+  )
+
+task(TASK_CHUGSPLASH_APPROVE)
+  .setDescription('Allows a manager to approve a bundle to be executed.')
+  .addParam('bundleHash', 'hash of the bundle')
+  .addParam('deployConfig', 'path to chugsplash deploy config')
+  .setAction(
+    async (
+      args: {
+        bundleHash: string
+        deployConfig: string
+      },
+      hre
+    ) => {
+      spinner.start('Approving the bundle...')
+
+      const config: ChugSplashConfig = await hre.run(TASK_CHUGSPLASH_LOAD, {
+        deployConfig: args.deployConfig,
+      })
+
+      const provider = new providers.JsonRpcProvider()
+      const ChugSplashRegistry = new Contract(
+        CHUGSPLASH_REGISTRY_ADDRESS,
+        ChugSplashRegistryABI,
+        await provider.getSigner()
+      )
+
+      await ChugSplashRegistry.approveChugSplashBundle(
+        config.options.name,
+        args.bundleHash
+      )
+      spinner.succeed('Bundle successfully approved')
+    }
+  )
+
+// TODO: Exclude listing bundles that have been executed
```

Comment:
Alternatively, maybe we can do a `--include-executed` boolean flag

Reply:
I updated this task so that it displays bundles as either proposed, approved, or executed (the latter not visible by default)

Reply:
Ok perfect